src/rez/config.py

```diff
@@ -11,13 +11,21 @@
 from rez.vendor.schema.schema import Schema, SchemaError, And, Or, Use
 from rez.vendor import yaml
 from rez.vendor.yaml.error import YAMLError
+from rez.vendor.six import six
 from rez.backport.lru_cache import lru_cache
 from contextlib import contextmanager
 from inspect import ismodule
 import os
+import sys
 import os.path
 import copy
 
+PY3 = sys.version_info >= (3, 0, 0)
+if PY3:
+    str_type = str
+else:
+    str_type = str
```

Comment:
This would prohibit unicode strings in PY2, which would be backwards incompatible, right? Is that really the intention?

Reply:
Yes, that didn't look right; it used to be basestring but looks to have been changed to str, despite the if-statement.

I would expect tests to fail if any backwards incompatibilities are introduced, and in this case no tests did fail. But still, better safe than sorry.